ADR/0022-secret-mgmt-for-user-workloads.md

```diff
@@ -26,65 +29,142 @@ Accepted
 
-#### Example: If the destination is `Environment`
+
+
+#### Example: If the working Secret should be created along with RemoteSecret.
 ```yaml
-apiVersion: v1
-kind: Secret
+apiVersion: appstudio.redhat.com/v1beta1
+kind: RemoteSecret
 metadata:
-  name: tmp-jdbc-connection-parameter-upload
-  labels:
-    spi.appstudio.redhat.com/upload-secret: secret
-    spi.appstudio.redhat.com/target-environment: prod
-    spi.appstudio.redhat.com/target-secret-name: jdbc-connection-parameter
-type: Opaque
-data:
- ...
+    namespace: jdoe-tenant
+spec:
+    target:
+        - namespace: spi-test-target
+          secretName: test-remote-secret-secret-5bmq9
+status:
+    TBD
 ```
+> :warning: **Note for this and following:** `RemoteSecret` is not Ready, `SecretData` referenced to it has to be uploaded. See `Uploading secret data to RemoteSecret` example.
+
+#### Example: If RemoteSecret has to be created and uploaded without setting any target
+
 ```yaml
 apiVersion: appstudio.redhat.com/v1beta1
 kind: RemoteSecret
 metadata:
-  name: jdbc-connection-parameter-prod
+    namespace: jdoe-tenant
 spec:
-  secret:
-    environment: prod
-    name: jdbc-connection-parameter
- ...
+    target:
+status:
+  TBD
 ```
 
-#### Example: If the destination is `Namespace` and secret has to be linked to SA
+#### Example:If multiple working Secrets should be created along with RemoteSecret.
 ```yaml
-apiVersion: v1
-kind: Secret
+apiVersion: appstudio.redhat.com/v1beta1
+kind: RemoteSecret
 metadata:
-  name: jdbc-connection-parameter
-  annotations:
-      spi.appstudio.redhat.com/secret-linkedto-serviceAccount-name: sa-star
-  labels:
-    spi.appstudio.redhat.com/upload-secret: secret
-    spi.appstudio.redhat.com/target-namespace: ns-9459039345
-type: Opaque
-data:
- ...
+    name: test-remote-secret-secret
+spec:
+    target:
+        - namespace: jdoe-test-ns
+          secretName: test-remote-secret-secret-5bmq9
+        - namespace: jdoe-prod-ns
+          secretName: test-remote-secret-secret-5bmq9
+status:
+    TBD
+```
+
+
+#### Example: If RemoteSecret has to be created with multiple target namespaces including BYC use-case
+```yaml
+apiVersion: appstudio.redhat.com/v1beta1
+kind: RemoteSecret
+metadata:
+    name: test-remote-secret-secret
+spec:
+    target:
+        - namespace: jdoe-test-ns
+          secretName: test-remote-secret-secret-5bmq9
+        - namespace: jdoe-prod-ns
+          secretName: test-remote-secret-secret-5bmq9
+          apiURL: https://somedomain.copm:443
+          clusterCredentialsSecret: team-a--prod-dtc--secret
+status:
+    TBD
 ```
+
+#### Example: If RemoteSecret has to be created with target namespace and Environment
+```yaml
+apiVersion: appstudio.redhat.com/v1beta1
+kind: RemoteSecret
+metadata:
+    name: test-remote-secret-secret
+    labels:
+        appstudio.redhat.com/remotesecret-target-environment: prod
```

Comment:
Can we remove "remotesecret-" leaving just appstudio.redhat.com/target-environment for ex.
I believe the fact it is belongs to RemoteSecret is enough?

Reply:
it's there just to say that this info is meant for remote secrets. it could get confused with deployment target which I believe doesn't overlap with this?

Reply:
Yes, it is definitely for remotesecret b/c it is part of its metadata (it's RemoteSecret's label)

In fact I do not think "-target" is bringing a lot of information here as well. 
IMO, the same as for remotesecret-, it just makes the name longer for nothing.
It even may act as additional point of confusion for users. Talking RHTAP terminology it is JUST environment/component/application. Prefix target- may make one think that it is something else.